vmm_tests/vmm_tests/tests/tests/x86_64/openhcl_linux_direct.rs

```diff
@@ -385,26 +385,58 @@ async fn storvsp(config: PetriVmBuilder<OpenVmmPetriBackend>) -> Result<(), anyh
 
     let sh = agent.unix_shell();
     // The drive ordering is not guaranteed, so we need to check all drives.
-    let output = cmd!(sh, "sh -c 'cat /sys/block/sd*/size'").read().await?;
+    let cmd = r#"
+exec 2>&1;
+for device in /sys/block/sd*; do
+    dev_name=$(basename "$device")
+    size_in_sectors=$(cat "$device/size")
+    echo "/dev/$dev_name $size_in_sectors"
+done
+"#;
+    // Run the enumeration script and log stdout+stderr
+    let output = sh.cmd("sh").arg("-c").arg(cmd).read().await?;
+    tracing::info!("guest cmd: enumerate /sys/block/sd* ->\n{}", output);
     // Make sure the disk sizes match.
     let reported_sizes = output
-        .split_ascii_whitespace()
-        .map(|x| x.parse::<u64>())
+        .split_terminator('\n')
+        .map(|x| -> Result<_, anyhow::Error> {
+            let line = x.split_ascii_whitespace().collect::<Vec<_>>();
+            if line.len() != 2 {
+                anyhow::bail!("unexpected line format: {}", x);
+            }
+            let size = line[1].parse::<u64>().context("failed to parse size")?;
+            let device = line[0];
```

Comment:
```suggestion
            let line = x.split_ascii_whitespace().collect::<Vec<_>>();
            let &[device, size] = &line else {
                anyhow::bail!("unexpected line format: {x}");
            };
            let size = size.parse::<u64>().context("failed to parse size")?;
```

Reply:
Thanks, that's much cleaner.

Reply:
(Vec<&str> can't be implicitly converted to a slice; the correct incantation is `&[device, size] = &line[..]`).